### PR TITLE
python provider: support mysql backends for django

### PR DESCRIPTION
**Pull request by char8**
Fixes https://github.com/railwayapp/nixpacks/issues/483 and Fixes https://github.com/railwayapp/nixpacks/issues/488 by providing the dependency required by the `mysqlclient` python package.

Tested building the example linked in https://github.com/railwayapp/nixpacks/issues/488 (🙏 ) locally and running the resulting container against a test PlanetScale DB. Assuming it works the same for a generic MySQL/MariaDB server.

### CODE REVIEW

**Comment by coffee-cup** [comment]
This is awesome! Can you please add a test + example app that uses Django and mysql. For tests, I think we should have
- Unit test in the Python provider to check that `is_using_mysql` works
- Generate plan test (generated automatically) to check that the build plan contains the correct package (https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md#snapshot-tests)
- Docker run test to check that the container can be built and run if it uses MySQL (can see how the Postgres Django test works)

File: tests/docker_run_tests.rs [hunk 1]
```diff
@@ -245,6 +247,63 @@ fn run_postgres() -> Container {
...
+        .spawn()
+        .unwrap()
+        .wait()
+        .context("starting mysql")
+        .unwrap();
+
+    // MySQL starts listening for connections after it has initialised its default database
+    // give it enough time to start before we continue with starting the application
+    // TODO: swap this for a healthceck + retry-loop because the arbitrary delay can lead to flakey tests
+    thread::sleep(Duration::new(10, 0));
```
Comment (char8) [inline tests/docker_run_tests.rs:290]: @coffee-cup thanks for the feedback - implemented the tests. But MySQL is being a bit annoying here, will address the TODO and ping you once it's ready for review.

Thinking of running a retry-loop with timeout (say 30s) of a command to check if MySQL is ready or not inside this function.

File: examples/python-django-mysql/mysite/settings.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+WSGI_APPLICATION = 'mysite.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/4.0/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.mysql',
```
Comment (char8) [inline examples/python-django-mysql/mysite/settings.py:78]: Copy of the django-postgres example with this line and L4 of requirements.txt updated

File: tests/docker_run_tests.rs (hunk 1 above)
Comment (char8) [reply tests/docker_run_tests.rs:290]: actually - looking at the runtimes, it seems like some of the docker tests take a few mins to complete. Could also add some margin to the sleep  (eg: make it 20s) and leave it as-is to reduce complexity if we're ok with that.

Comment (char8) [reply tests/docker_run_tests.rs]: whoops missed the unit test

**Review by coffee-cup — APPROVED** [review]
Awesome! 🎉

**Pull request merged**